From ee17068e5dc61ccb4769830b7ac8c61833cb429b Mon Sep 17 00:00:00 2001
From: Bartosz Bilas <bartosz.bilas@hotmail.com>
Date: Sun, 27 Oct 2019 09:01:12 +0100
Subject: [PATCH] stm32mp1: configs: fix checking the presence of an
 environment

Execute env check command within extra env settings section instead of
bootcmd whereby we are able to mount rootfs partition from sd card properly.

Signed-off-by: Bartosz Bilas <bartosz.bilas@hotmail.com>
---
 include/configs/stm32mp1.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/stm32mp1.h b/include/configs/stm32mp1.h
index 42717c167e..ffa1cbc954 100644
--- a/include/configs/stm32mp1.h
+++ b/include/configs/stm32mp1.h
@@ -146,7 +146,6 @@
 	"if test ${boot_device} = serial || test ${boot_device} = usb;" \
 	"then stm32prog ${boot_device} ${boot_instance}; " \
 	"else " \
-		"run env_check;" \
 		"if test ${boot_device} = mmc;" \
 		"then env set boot_targets \"mmc${boot_instance}\"; fi;" \
 		"if test ${boot_device} = nand;" \
@@ -223,6 +222,7 @@
 	"ramdisk_addr_r=0xc4400000\0" \
 	"altbootcmd=run bootcmd\0" \
 	"env_default=1\0" \
+	"run env_check;" \
 	"env_check=if test $env_default -eq 1;"\
 		" then env set env_default 0;env save;fi\0" \
 	STM32MP_BOOTCMD \
-- 
2.34.1

